jupyterhub/apihandlers/base.py

```diff
@@ -31,6 +31,12 @@ def check_referer(self):
         if not referer:
             self.log.warning("Blocking API request with no referer")
             return False
+
+        # Checking CORS
+        allow_origin = self.settings.get('allow_origin', None)
+        self.log.warning(allow_origin)
+        if (allow_origin == '*') or (allow_origin and allow_origin in referer_path):
```

Comment:
Thanks for your feedback.
To make the story short, I am trying to run JupyterHub behind a frontend. When doing so, clicking the butting "Stop my server" fails because of X-Origin.